packages/graphing/controller/src/index.js

```diff
@@ -0,0 +1,338 @@
+import debug from 'debug';
+
+import cloneDeep from 'lodash/cloneDeep';
+import lodash from 'lodash';
+import isEqual from 'lodash/isEqual';
+import { tools } from '@pie-lib/graphing';
+
+const { utils } = tools;
+const { sinY, buildDataPoints, getAmplitudeAndFreq, FREQ_DIVIDER, parabolaFromTwoPoints } = utils;
+
+const log = debug('@pie-element:graphing:controller');
+
+export const equalPoint = (A, B) => {
+  // x1 = x2 & y1 = y2
+  let equalLabel = true;
+
+  if (A.label || B.label) {
+    equalLabel = isEqual(A.label, B.label);
+  }
+
+  return isEqual(A.x, B.x) && isEqual(A.y, B.y) && equalLabel;
+};
+
+export const equalSegment = (A, B, C, D) => {
+  // x1 = x3 & y1 = y3 & x2 = x4 & y2 = y4
+  return ((isEqual(A, C) && isEqual(B, D)) || ((isEqual(B, C) && isEqual(A, D))));
+};
+
+export const equalVector = (A, B, C, D) => {
+  // x1 = x3 & y1 = y3 & x2 = x4 & y2 = y4
+  return ((isEqual(A, C) && isEqual(B, D)));
+};
+
+export const equalLine = (A, B, C, D) => {
+  // (y2 - y1)/(x2 - x1) = (y4 - y3)/(x4 - x3);
+  return (((B.y - A.y) / (B.x - A.x)) === ((D.y - C.y) / (D.x - C.x)));
+};
+
+export const equalRay = (A, B, C, D) => {
+  // line & x1 = x3 & y1 = y3 & angle between (x1, y1) (x2, y2) is same as angle between (x3, y3) (x4, y4)
+  return ((
+        (B.y - A.y) / (B.x - A.x)
+      ) === (
+      (D.y - C.y) / (D.x - C.x))
+    ) && (A.x === C.x && A.y === C.y) &&
+    (
+      Math.atan2(B.y - A.y, B.x - A.x) * 180 / Math.PI ===
+      Math.atan2(D.y - C.y, D.x - C.x) * 180 / Math.PI
+    );
+};
+
+export const equalPolygon = (pointsA, pointsB) => {
+  const sessAnswerPoints = lodash.uniqWith(pointsA, isEqual);
+  const withoutDuplicates = lodash.uniqWith(pointsB, isEqual);
+  const sB = lodash.orderBy(sessAnswerPoints, ['x','y'], ['asc', 'asc']);
+  const sD = lodash.orderBy(withoutDuplicates, ['x','y'], ['asc', 'asc']);
+
+  return isEqual(sD, sB);
+};
+
+export const equalCircle = (A, B, C, D) => {
+  const equalRootAndEdge = isEqual(D, B) && isEqual(C, A);
+  const rAB = Math.sqrt(((B.x - A.x) ** 2) + ((B.y - A.y) ** 2));
+  const rCD = Math.sqrt(((D.x - C.x) ** 2) + ((D.y - C.y) ** 2));
+  const equalRAndRoot = isEqual(C, A) && isEqual(rAB, rCD);
+
+  return equalRootAndEdge || equalRAndRoot;
+};
+
+export const equalSine = (A, B, C, D) => {
```

Comment:
there is probably a mathematical way to compare the 2 sine waves. .. 1. you'll need to derive the equation from 2 points, then compare the equation for each